verifier/verifier.go

```diff
@@ -898,3 +903,135 @@ func logVerificationResult(logger log.Logger, result *notation.ValidationResult)
+	if len(signerInfo.UnsignedAttributes.TimestampSignature) == 0 {
+		return errors.New("no timestamp countersignature was found in the signature envelope")
+	}
+
+	// 2. Verify the timestamp countersignature
+	logger.Info("Verifying the timestamp countersignature...")
+	signedToken, err := tspclient.ParseSignedToken(signerInfo.UnsignedAttributes.TimestampSignature)
+	if err != nil {
+		return fmt.Errorf("failed to parse timestamp countersignature with error: %w", err)
+	}
+	info, err := signedToken.Info()
+	if err != nil {
+		return fmt.Errorf("failed to get the timestamp TSTInfo with error: %w", err)
+	}
+	trustTSACerts, err := loadX509TSATrustStores(ctx, outcome.EnvelopeContent.SignerInfo.SignedAttributes.SigningScheme, policyName, trustStores, x509TrustStore)
+	if err != nil {
+		return fmt.Errorf("failed to load tsa trust store with error: %w", err)
+	}
+	if len(trustTSACerts) == 0 {
+		return errors.New("no trusted TSA certificate found in trust store")
+	}
+	rootCertPool := x509.NewCertPool()
+	for _, trustedCerts := range trustTSACerts {
+		rootCertPool.AddCert(trustedCerts)
+	}
+	timestamp, err := info.Validate(signerInfo.Signature)
+	if err != nil {
+		return fmt.Errorf("failed to get timestamp from timestamp countersignature with error: %w", err)
+	}
+	tsaCertChain, err := signedToken.Verify(ctx, x509.VerifyOptions{
+		CurrentTime: timestamp.Value,
+		Roots:       rootCertPool,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to verify the timestamp countersignature with error: %w", err)
+	}
+
+	// 3. Validate timestamping certificate chain
+	logger.Info("Validating timestamping certificate chain...")
+	if err := nx509.ValidateTimestampingCertChain(tsaCertChain); err != nil {
+		return fmt.Errorf("failed to validate the timestamping certificate chain with error: %w", err)
+	}
+	logger.Info("TSA identity is: ", tsaCertChain[0].Subject)
+
+	// 4. Perform the timestamping certificate chain revocation check
+	logger.Info("Checking timestamping certificate chain revocation...")
+	certResults, err := r.Validate(tsaCertChain, time.Time{})
+	if err != nil {
+		return fmt.Errorf("failed to check timestamping certificate chain revocation with error: %w", err)
+	}
+	finalResult, problematicCertSubject := revocationFinalResult(certResults, tsaCertChain, logger)
+	switch finalResult {
+	case revocationresult.ResultOK:
+		logger.Debug("No verification impacting errors encountered while checking timestamping certificate chain revocation, status is OK")
+	case revocationresult.ResultRevoked:
+		return fmt.Errorf("timestamping certificate with subject %q is revoked", problematicCertSubject)
+	default:
+		// revocationresult.ResultUnknown
+		return fmt.Errorf("timestamping certificate with subject %q revocation status is unknown", problematicCertSubject)
+	}
+
+	// 5. Check the timestamp against the signing certificate chain
+	logger.Info("Checking the timestamp against the signing certificate chain...")
```

Comment:
Aren't we combining debug logs with info logs?

Reply:
Updated.

Reply:
Changed most of them to debug. But kept the two logs stating that `Timestamp verification disabled` as `Info` level. Because they are high level messages that's related to the user's trust policy.